scripts/validate_docstrings.py

```diff
@@ -490,10 +491,41 @@ def yields(self):
     @property
     def method_source(self):
         try:
-            return inspect.getsource(self.obj)
+            source = inspect.getsource(self.obj)
+            # Remove common indentation.
+            first_spaces = re.match(' +', source)
+            if first_spaces:
+                first_spaces = first_spaces.group(0)
+                source = re.sub('^' + first_spaces, '', source)
+                source = re.sub(r'\n' + first_spaces, r'\n', source)
+            return source
         except TypeError:
             return ''
 
+    @property
+    def method_returns(self):
+        tree = ast.parse(self.method_source).body
+        if tree:
+            root = tree[0]
+
+            def gather_returns(node):
```

Comment:
move the function definition at the beginning of the parent function, it makes things clearer.